tests/foreman/cli/test_content_credentials.py

```diff
@@ -0,0 +1,880 @@
+# -*- encoding: utf-8 -*-
+"""Test class for GPG Key CLI.
+The gpg sub-command was deprecated in favour of content-credential in
+Satellite 6.8
+
+:Requirement: ContentCredential
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ContentCredentials
+
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from tempfile import mkstemp
+
+from fauxfactory import gen_alphanumeric
+from fauxfactory import gen_choice
+from fauxfactory import gen_integer
+from fauxfactory import gen_string
+
+from robottelo import ssh
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.content_credentials import ContentCredential
+from robottelo.cli.factory import CLIFactoryError
+from robottelo.cli.factory import make_content_credential
+from robottelo.cli.factory import make_org
+from robottelo.cli.factory import make_product
+from robottelo.cli.factory import make_repository
+from robottelo.cli.org import Org
+from robottelo.cli.product import Product
+from robottelo.cli.repository import Repository
+from robottelo.constants import DEFAULT_ORG
+from robottelo.constants import VALID_GPG_KEY_FILE
+from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import valid_data_list
+from robottelo.decorators import tier1
+from robottelo.decorators import tier2
+from robottelo.decorators import upgrade
+from robottelo.helpers import get_data_file
+from robottelo.test import CLITestCase
+
+VALID_GPG_KEY_FILE_PATH = get_data_file(VALID_GPG_KEY_FILE)
+
+
+def create_gpg_key_file(content=None):
+    """Creates a fake GPG Key file and returns its path or None if an error
+    happens.
+    """
+
+    (_, key_filename) = mkstemp(text=True)
+    if not content:
+        content = gen_alphanumeric(gen_integer(20, 50))
+    with open(key_filename, "w") as gpg_key_file:
+        gpg_key_file.write(content)
+        return key_filename
+
+    return None
+
+
+class TestGPGKey(CLITestCase):
+    """Tests for GPG Keys via Hammer CLI"""
+
+    search_key = 'name'
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a shared organization for all tests to avoid generating
+        hundreds of organizations
+        """
+        super(TestGPGKey, cls).setUpClass()
+        cls.org = make_org(cached=True)
+
+    # Bug verification
+
+    @tier1
+    def test_verify_redmine_4272(self):
```

Comment:
I don't like this pattern of the test function name reflecting a BZ or issue tracker number.  The test function name should provide some context of what is being tested. The issue that is related to the test should be part of its metadata.

Reply:
@mshriver it is extremely odd, agree its not good (it came from gpg key file). Will change it to more suitable name.

Reply:
Fixed. 

BTW, this is the redmine issue: https://projects.theforeman.org/issues/4272